### ISSUE TITLE
Geographic raster resource creation fails if the geotiff file has some metadata missing

### ISSUE THREAD

**Issue by dtarb** [hydroshare/hydroshare#77]
The following error results if a geographic raster resource is created that does not have projection metadata.

![capture](https://cloud.githubusercontent.com/assets/3456833/5793338/aea064e6-9f0d-11e4-9d93-6516b16d8d6b.JPG)

The file that generates this error is at http://beta.hydroshare.org/r/32ae0f96058e4651bde9095166525f65/ as a generic resource.

HydroShare should gracefully handle a file with some missing metadata by indicating that the field is missing and providing capability to add/enter it.  See #75.

**Comment by hyi** [comment]
This involves two sub-issues. First, this error was caused by a Django file upload default behavior. Specifically, by default, Django holds the uploaded file in memory if the uploaded file size is less than 2.5mb, otherwise, django writes uploaded file to a temporary location on disk. In this case, the uploaded geotiff file size is ~850K, which is less than 2.5mb, hence the file was held in memory and can only be read via django class from memory. However, gdal lib we used to extract metadata from geotiff needs to open a file from disk, hence the error. To resolve this issue, I override Django's default file upload behavior by setting FILE_UPLOAD_MAX_MEMORY_SIZE to be zero, hence forcing Django to always write uploaded file to disk rather than holding it in memory. This change was made under consideration that other lib functions we use to extract metadata for various resource types do not have interface to directly load file from memory held up by Django for small uploaded files, and the performance gain for storing small uploaded files in memory is not so great for our use case. The second sub-issue is to handle missing metadata fields from the uploaded file, which has now been fixed to hold up "Unnamed" if corresponding metadata fields have missing values for users to enter (in current resource creation workflow) or edit later (in the new resource creation workflow to be implemented). Fixes pushed to github to be merged into develop branch.

**Comment by dtarb** [comment]
@hyi I still get an error when loading the mawhefel.tif file.  This time a different error.
![capture](https://cloud.githubusercontent.com/assets/3456833/6574546/a450549e-c6fc-11e4-93a8-4f4826b9c34a.PNG)

The file that triggers this is at http://beta.hydroshare.org/resource/a0f2108074444ff3a9480c33eb39c106/

**Comment by dtarb** [comment]
Reopened - I pushed the wrong button

**Comment by hyi** [comment]
@dtarb I got a 403 forbidden when accessing the resource link on beta, so I am not able to see the exact error with this particular resource. But looking at the error you pasted here, it looks like it comes from the metadata extraction function that uses osr.py that is written by Tian. So Tian might need to look into this issue once we can see the exact error with the test data you are using.

**Comment by dtarb** [comment]
You should be able to see the resource link on beta now.  I had 
forgotten to make public.

Dave

**Comment by hyi** [comment]
This issue is caused since ors.CoordinateTransformation() returns None. Fixed this issue by adding check for None and use original coverage if wgs84 coverage cannot be obtained due to missing coordinate system info or other issues. Will post the fix to this issue into the 186-branch which I am currently working on for comments to be merged together into develop soon when ready.

**Comment by hyi** [comment]
Fix pushed to github and a pull request has been created to merge it into develop.

**Comment by dtarb** [comment]
Fixed in alpha 1.4.0 rc0
